src/Sulu/Bundle/MediaBundle/Controller/MediaController.php

```diff
@@ -211,6 +211,99 @@ public function deleteAction($id)
     }
 
     /**
+     * Creates a new preview image and saves it to the provided media.
+     *
+     * @Post("media/preview/{id}")
+     *
+     * @param int $id
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \Sulu\Bundle\MediaBundle\Media\Exception\CollectionNotFoundException
+     */
+    public function postPreviewAction($id, Request $request)
+    {
+        try {
+            $mediaManager = $this->getMediaManager();
+            $systemCollectionManager = $this->get('sulu_media.system_collections.manager');
+
+            $locale = $this->getLocale($request);
+
+            $media = $mediaManager->getById($id, $locale);
+            $mediaEntity = $media->getEntity();
+
+            $data = $this->getData($request, false);
+            unset($data['id']);
+
+            if ($mediaEntity->getPreviewImage() !== null) {
+                $data['id'] = $mediaEntity->getPreviewImage()->getId();
+            }
+            $data['collection'] = $systemCollectionManager->getSystemCollection('sulu_media.preview_img');
+            $data['locale'] = $locale;
+            $data['title'] = $media->getTitle();
+
+            $uploadedFile = $this->getUploadedFile($request, 'previewImg');
+            $previewImg = $mediaManager->save($uploadedFile, $data, $this->getUser()->getId());
+
+            $mediaEntity->setPreviewImage($previewImg->getEntity());
+            $mediaManager->saveEntity($mediaEntity);
+
+            // Add preview thumbnails
+            $media = $mediaManager->addFormatsAndUrl($media);
+
+            $view = $this->view($media, 200);
+        } catch (MediaNotFoundException $e) {
+            $view = $this->view($e->toArray(), 404);
+        } catch (MediaException $e) {
+            $view = $this->view($e->toArray(), 400);
+        }
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * Removes current preview image and sets default video thumbnail.
+     *
+     * @Get("media/reset-preview/{id}")
```

Comment:
A get should never delete something... This should also go in the child resource I've mentioned in the previous method. Then it can be a delete action.